rules/no-for-loop.js

```diff
@@ -304,9 +323,15 @@ const create = context => {
 							node.init.range[0],
 							node.update.range[1]
 						], replacement),
-
-						fixer.removeRange(getRemovalRange(elementVariableDeclaration, sourceCode))
-					];
+						...arrayReferences.map(reference => {
+							if (reference === elementReference) {
+								return null;
+							}
+
+							return fixer.replaceText(reference.identifier.parent, element);
+						}),
+						elementNode ? fixer.removeRange(getRemovalRange(elementNode, sourceCode)) : null
```

Comment:
I prefer using `undefined`.

```suggestion
						elementNode && fixer.removeRange(getRemovalRange(elementNode, sourceCode))
```